radicle-node/src/service/message.rs

```diff
@@ -173,46 +166,119 @@ impl wire::Decode for NodeAnnouncement {
     }
 }
 
+/// Node announcing project refs being created or updated.
 #[derive(Debug, Clone, PartialEq, Eq)]
 pub struct RefsAnnouncement {
     /// Repository identifier.
     pub id: Id,
     /// Updated refs.
     pub refs: Refs,
+    // TODO: Add timestamp
+}
+
+/// Node announcing its inventory to the network.
+/// This should be the whole inventory every time.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct InventoryAnnouncement {
+    pub inventory: Vec<Id>,
+    pub timestamp: Timestamp,
+}
+
+/// Announcement messages are messages that are relayed between peers.
+#[derive(Clone, PartialEq, Eq)]
+pub enum AnnouncementMessage {
+    /// Inventory announcement.
+    Inventory(InventoryAnnouncement),
+    /// Node announcement.
+    Node(NodeAnnouncement),
+    /// Refs announcement.
+    Refs(RefsAnnouncement),
 }
 
-impl RefsAnnouncement {
-    /// Verify a signature on this message.
-    pub fn verify(&self, signer: &NodeId, signature: &crypto::Signature) -> bool {
-        let msg = wire::serialize(self);
-        signer.verify(&msg, signature).is_ok()
+impl AnnouncementMessage {
+    /// Sign this announcement message.
+    pub fn signed<S: crypto::Signer>(self, signer: S) -> Announcement {
```

Comment:
Was there a reason to change the method to past tense?

Reply:
All methods that are "functional" in that they move self and return a "changed" type use the past tense. So eg. in this case, the method doesn't sign `AnnouncementMessage`, it returns a signed announcement.